commands/audit/jasrunner.go

```diff
@@ -2,62 +2,119 @@ package audit
 		return
 	}
-	scanner, err := jas.NewJasScanner(workingDirs, serverDetails)
+	scanner, err := jas.NewJasScanner(serverDetails, jfrogAppsConfig)
 	if err != nil {
 		return
 	}
+	// Set environments variables for analytics in analyzers manager.
+	callback := jas.SetAnalyticsMetricsDataForAnalyzerManager(msi, scanResults.GetScaScannedTechnologies())
 	defer func() {
+		auditParallelRunner.ScannersWg.Wait()
+		callback()
 		cleanup := scanner.ScannerDirCleanupFunc
 		err = errors.Join(err, cleanup())
 	}()
-	if progress != nil {
-		progress.SetHeadlineMsg("Running applicability scanning")
-	}
-	// Set environments variables for analytics in analyzers manager.
-	callback := jas.SetAnalyticsMetricsDataForAnalyzerManager(msi, scanResults.GetScaScannedTechnologies())
-	defer callback()
-	scanResults.ExtendedScanResults.ApplicabilityScanResults, err = applicability.RunApplicabilityScan(scanResults.GetScaScansXrayResults(), directDependencies, scanResults.GetScaScannedTechnologies(), scanner, thirdPartyApplicabilityScan)
-	if err != nil {
-		return
-	}
+
 	// Don't execute other scanners when scanning third party dependencies.
-	if thirdPartyApplicabilityScan {
-		return
+	if !auditParams.thirdPartyApplicabilityScan {
+		for _, module := range scanner.JFrogAppsConfig.Modules {
+			if !jas.ShouldSkipScanner(module, utils.Secrets) {
+				auditParallelRunner.ScannersWg.Add(1)
+				_, err = auditParallelRunner.Runner.AddTaskWithError(runSecretsScan(auditParallelRunner, scanner, scanResults, module), auditParallelRunner.AddErrorToChan)
+				if err != nil {
+					return fmt.Errorf("failed to create secrets scan task: %s", err.Error())
+				}
+			}
+			if !jas.ShouldSkipScanner(module, utils.IaC) {
+				auditParallelRunner.ScannersWg.Add(1)
+				_, err = auditParallelRunner.Runner.AddTaskWithError(runIacScan(auditParallelRunner, scanner, scanResults, module), auditParallelRunner.AddErrorToChan)
+				if err != nil {
+					return fmt.Errorf("failed to create iac scan task: %s", err.Error())
+				}
+			}
+			if !jas.ShouldSkipScanner(module, utils.Sast) {
+				auditParallelRunner.ScannersWg.Add(1)
+				_, err = auditParallelRunner.Runner.AddTaskWithError(runSastScan(auditParallelRunner, scanner, scanResults, module), auditParallelRunner.AddErrorToChan)
+				if err != nil {
+					return fmt.Errorf("failed to create sast scan task: %s", err.Error())
+				}
+			}
+		}
 	}
-	if progress != nil {
-		progress.SetHeadlineMsg("Running secrets scanning")
+
+	// Wait for sca scan to complete
+	auditParallelRunner.ScaScansWg.Wait()
+	for _, module := range scanner.JFrogAppsConfig.Modules {
+		if !jas.ShouldSkipScanner(module, utils.Applicability) {
+			auditParallelRunner.ScannersWg.Add(1)
+			_, err = auditParallelRunner.Runner.AddTaskWithError(runContextualScan(auditParallelRunner, scanner, scanResults, module, auditParams), auditParallelRunner.AddErrorToChan)
+			if err != nil {
+				return fmt.Errorf("failed to create contextual scan task: %s", err.Error())
+			}
+		}
```

Comment:
```suggestion
    if err = addModuleJasScanTask(module, utils.Applicability, auditParallelRunner, runContextualScan(auditParallelRunner, scanner, scanResults, module, auditParams)); err != nil {
        return
    }
```
can be used here too